moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -213,6 +229,18 @@ def get_attached_objects(self, object_ids=[]):
             conversions.msg_from_string(msg, ser_aobjs[key])
             aobjs[key] = msg
         return aobjs
+    
+    def apply_collision_object(self, collision_object_message):
+        """
+        Applies the collision object message to the scene.
+        """
+        return self._psi.apply_collision_object(conversions.msg_to_string(collision_object_message))
+    
+    def apply_planning_scene(self, planning_scene_message):
+        """
+        Applies the planning scene message.
+        """
+        return self._psi.apply_planning_scene(conversions.msg_to_string(planning_scene_message))
```

Comment:
I don't think we should add these methods.

[`__submit` will already use the service](https://github.com/ros-planning/moveit/pull/2532/files#diff-ea03d418c2c0d6b5328a50bd314ece97aa036b872aaa8d04f37c82c3490a0fb3R78) iff the PSI object is instantiated as [`synchronous`](https://github.com/ros-planning/moveit/pull/2532/files#diff-ea03d418c2c0d6b5328a50bd314ece97aa036b872aaa8d04f37c82c3490a0fb3R71).

There is no need or merit in depending on the C++ implementation here.

Reply:
`__submit` is not user-facing, so users of this class can't publish their own `CollisionObject`. 

I also don't see the problem with depending on the C++ implementation.

I can implement `apply_collision_object` and `apply_attached_collision_object` with a call to `__submit`, but a pure Python API doesn't seem like it will be maintained, so adding C++ bindings with familiar names seems better.

Reply:
> __submit is not user-facing, so users of this class can't publish their own CollisionObject.

That's true, but that's why you added the `add_object` method that *uses* `__submit` (just like almost every other method in this class)!
So you basically add a second way to do the same thing if you also expose the C++ `apply*` methods.

Reply:
Oh. Right. These commits are so old that I forgot that I did that. I just knew that they worked and we've been using them.

I'm not sure what I think about 1) the naming being different from C++ for no reason and 2) maintaining a pure Python API instead of bindings as described in #900, but I can leave out the C++ methods for now.

Are you good with `apply_collision_object` and `apply_attached_collision_object` calling `__submit`?

Reply:
> I'm not sure what I think about 1) the naming being different from C++ for no reason and

neither am I, but that's what we got. The idea back then was to avoid breaking the whole class, but still support the update service for all existing methods. The only proper way to have a 1-to-1 mapping with the C++ classes would be to create an entirely new package for it. It's an interesting idea, but quite some work too. :-)

> Are you good with apply_collision_object and apply_attached_collision_object calling __submit?

No, because if `synchronous == False`, these methods would use the topic interfaces then!
With the current design we should not have the add vs apply distinction we have in C++.

Reply:
Also true.

Alright, I took `applyCollisionObject` out. `applyPlanningScene` was there already though, it just wasn't exposed in Python. Surely that was an oversight? I don't see a way to publish a PlanningScene message synchronously otherwise.

Yes, more Python bindings would be great. Someone from #900 should start adding PlanningScene (not PSI) bindings for WMD.